### PR TITLE
Refactor CreateCustomerTokenService to use email and address object instead of order object.

### PR DESCRIPTION
**Pull request by piyushswain**
This commit changes the required arguments for the `CreateCustomerTokenService` class.

The required arguments before were: `order`, `authorization_token`, `region`.
The required arguments now are: `email`, `address`, `authorization_token`, `currency` and `region`.

This change was made due to requirements of the `CreateCustomerTokenService` class which has now been isolated.
This service needs to be used in places where there may not be an order to setup a subscription. In those cases we need to still create the request body and generate the customer_token.

Hence, the changes in this commit to remove the dependency from the order and instead just use the required information to generate the customer_token from Klarna.

### CODE REVIEW

File: lib/active_merchant/billing/klarna_gateway.rb [hunk 1]
```diff
@@ -260,11 +260,13 @@ def get_and_update_source(order_id)
 
       def create_profile(payment)
         return if payment.source.customer_token
-        return unless payment.order.klarna_tokenizable?
```
Comment (DanielePalombo) [inline lib/active_merchant/billing/klarna_gateway.rb:263]: why you have removed it?

File: spec/lib/active_merchant/billing/klarna_gateway_spec.rb [hunk 2]
```diff
@@ -180,19 +183,5 @@
         expect { create_profile }.not_to change(payment_source.reload, :customer_token).from(initial_customer_token)
       end
     end
-
-    context 'when Spree::Order#klarna_tokenizable? is false' do
```
Comment (DanielePalombo) [inline spec/lib/active_merchant/billing/klarna_gateway_spec.rb:184]: Why did you remove this test?

Comment (piyushswain) [reply spec/lib/active_merchant/billing/klarna_gateway_spec.rb:184]: I didn't remove it.

It got removed when I forked it and copied over the changes I had made.
I am looking for any more changes to the files I have modified right now.
Will add it again

File: lib/active_merchant/billing/klarna_gateway.rb (hunk 1 above)
Comment (piyushswain) [reply lib/active_merchant/billing/klarna_gateway.rb:263]: added it again

File: spec/lib/active_merchant/billing/klarna_gateway_spec.rb (hunk 2 above)
Comment (DanielePalombo) [reply spec/lib/active_merchant/billing/klarna_gateway_spec.rb]: oh, this is pretty dangerous. probably you have forked the repo before the PR that adds these changes was merged. Please update your master and rebase this branch again 🙏

**Review by DanielePalombo — APPROVED** [review]

**Review by DanielePalombo — APPROVED** [review]

**Review by Azeem838 — APPROVED** [review]

File: spec/lib/active_merchant/billing/klarna_gateway_spec.rb [hunk 3]
```diff
@@ -157,8 +158,10 @@
       create_profile
 
       expect(SolidusKlarnaPayments::CreateCustomerTokenService).to have_received(:call).with(
-        order: payment.order,
+        email: payment.order.email,
+        address: payment.order.billing_address || payment.order.shipping_address,
```
Comment (Naokimi) [inline spec/lib/active_merchant/billing/klarna_gateway_spec.rb:162]: you extracted `payment.order` into a variable in https://github.com/solidusio-contrib/solidus_klarna_payments/pull/34/files#diff-9bc32aaf55cd748994f6d15d831b1237fd3b8b22dd1bd20c958379a052aade3fR265 but not here - would be good to unify

**Review by Naokimi — APPROVED** [review]

File: spec/lib/active_merchant/billing/klarna_gateway_spec.rb (hunk 3 above)
Comment (DanielePalombo) [reply spec/lib/active_merchant/billing/klarna_gateway_spec.rb:162]: Sorry, I saw that comment after the merge. It doesn't look like blocking changes. If you think it needs to be addressed please open an issue or make your own PR.
